lib/logstash/plugin_mixins/common.rb

```diff
@@ -22,17 +22,30 @@ def setup_schema_registry_config
         # Option to set the proxy of the Schema Registry.
         # This option permits to define a proxy to be used to reach the schema registry service instance.
         config :schema_registry_proxy, :validate => :uri
+
+        # Option to skip validating the schema registry during registration. This can be useful when using
+        # certificate based auth
+        config :schema_registry_validation, :validate => ['auto', 'skip'], :default => 'auto'
       end
 
       def check_schema_registry_parameters
         if @schema_registry_url
           check_for_schema_registry_conflicts
           @schema_registry_proxy_host, @schema_registry_proxy_port  = split_proxy_into_host_and_port(schema_registry_proxy)
           check_for_key_and_secret
-          check_for_schema_registry_connectivity_and_subjects
+          check_for_schema_registry_connectivity_and_subjects if schema_registry_validation?
         end
       end
 
+      def schema_registry_validation?
+        return false if schema_registry_validation.to_s == 'skip'
+        !using_kerberos?
```

Comment:
Nitpick:

I have a personal preference for what I call "bail conditions" being consistently formatted to make them more readable at a glance. Here, we have two conditions that would cause us to skip the validation:
 - `schema_registry_validation => skip`; OR
 - kerberos authentication

I think it would read a bit more clearly to call these out individually and return `true` only if we have not bailed yet, instead of relying on the `!using_kereberos?` for the final line.

```suggestion
        return false if schema_registry_validation.to_s == 'skip'
        return false if using_kerberos? # pre-validation doesn't support kerberos
        
        true
```